pipelines/common/src/main/java/com/google/fhir/analytics/DwhFiles.java

```diff
@@ -315,4 +337,46 @@ public void writeToFile(String fileName, byte[] content) throws IOException {
       throw new IOException(errorMessage);
     }
   }
+
+  /**
+   * This method returns the file separator for the filesystem where the data-warehouse is hosted.
+   *
+   * @param dwhRootPrefix the root directory prefix of the filesystem.
+   * @return the file separator
+   */
+  public static String getFileSeparatorForDwhFiles(String dwhRootPrefix) {
+    String scheme = parseScheme(dwhRootPrefix);
+    switch (scheme) {
+      case DEFAULT_SCHEME:
+        return File.separator;
+      case GcsPath.SCHEME:
+        return "/";
+      default:
+        String errorMessage = String.format("File system scheme=%s is not yet supported", scheme);
+        log.error(errorMessage);
+        throw new IllegalArgumentException(errorMessage);
+    }
+  }
+
+  /**
+   * This method returns the schema of the passed in spec. This is inspired from the parsing logic
+   * in the repo <a href="https://github.com/apache/beam">Beam File System</a>
+   *
+   * @param spec - the spec from which the schema needs to be parsed
+   * @return the schema
+   */
+  public static String parseScheme(String spec) {
+    // The spec is almost, but not quite, a URI. In particular,
+    // the reserved characters '[', ']', and '?' have meanings that differ
+    // from their use in the URI spec. ('*' is not reserved).
+    // Here, we just need the scheme, which is so circumscribed as to be
+    // very easy to extract with a regex.
+    Matcher matcher = FILE_SCHEME_PATTERN.matcher(spec);
+
+    if (!matcher.matches()) {
+      return DEFAULT_SCHEME;
```

Comment:
I understand this is not introduced in this PR but I find relying on `FileSystems.DEFAULT_SCHEME` here confusing as it does not seem to have anything to do with Beam's `FileSystem.DEFAULT_SCHEME`. Instead it is just any arbitrary constant. Is my understanding correct? If yes, please simply introduce a new constant here in `DwhFiles` and remove dependence on `FileSystems.DEFAULT_SCHEME`.

Reply:
It is just used a mere constant, hence have made it as a constant